files/secure_media_views.py

```
@@ -100,6 +104,61 @@
+    Invalidate all cached file paths for a media.
+    Note: This requires pattern-based deletion (django-redis).
+    Returns number of keys deleted.
+    """
+    try:
+        if hasattr(cache, 'delete_pattern'):
+            # Can't easily clear by media_id without storing reverse mapping
+            # So we just let the cache expire naturally (5 min TTL)
+            # For critical updates, the cache TTL is short enough
+            logger.debug(f"Media path cache will expire naturally for media {media_id}")
+            return 0
+        return 0
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)